feat: add `on_activated`

tyush opened the pull request:
With this commit, applications are able to designate a callback that can be run when a notification is clicked.

Use cases for this include chat programs which may want to change the chat they display when a user taps a notification.

tyush committed:
Expose Activated event

Allows users to set an ToastNotification.Activated callback. Currently requires the callback to be 'static; this likely can be fixed by some lifetime and boxing trickery at the cost of readability.

tyush committed:
Show callback functionality in example.

tyush committed:
Revert "Show callback functionality in example."

This reverts commit 62fc7a0bc643d502b30749134792085d195cab85.

tyush committed:
Simplify callback in example.

May be a candidate for a new example instead of polluting the "simple" example.

amrbashir commented on src/lib.rs:
```diff
@@ -401,6 +406,16 @@ impl Toast {
         self
     }
 
+    // HACK: f is static so that we know the function is valid to call.
+    //       this would be nice to remove at some point
+    pub fn on_activated(
+        mut self,
+        f: impl FnMut(&Option<ToastNotification>, &Option<IInspectable>) -> Result<()> + Send + 'static,
+    ) -> Self {
+        self.on_activated = Some(TypedEventHandler::new(f));
+        self
+    }
```
There is no need to provide these arguments to the function, correct? I think we can just remove them and add them later if needed, also we could use `FnOnce` since that closure will only be activated once?

```suggestion
    // HACK: f is static so that we know the function is valid to call.
    //       this would be nice to remove at some point
    pub fn on_activated<F: FnOnce() -> Result<()> + Send + 'static>(
        mut self,
        f: F,
    ) -> Self {
        self.on_activated = Some(f);
        self
    }
```

you will also need to change `on_activated` signature

tyush replied:
As it turns out, the windows library adds the FnMut bound on all TypedEventHandlers. Should this be worked around, or kept in line with the library?

amrbashir replied:
Shame! lets use `FnMut` then

tyush committed:
Change FnMut requirement to FnOnce

Loosens the requirements on `on_activated.` Turns out I had my Fn trait hierarchies wrong :(

Co-authored-by: Amr Bashir <amr.bashir2015@gmail.com>

tyush committed:
Change callback back to FnMut, don't send params to callback

tyush committed:
Whoops that's not how closures move

tyush committed:
Remove callback params from simple.rs

amrbashir commented:
Looks like you need to run `cargo fmt`

tyush commented:
Whoops, formatted now.

tyush committed:
cargo fmt

amrbashir committed:
Create on-activated.md

amrbashir commented on examples/simple.rs:
```diff
@@ -3,14 +3,25 @@
-use tauri_winrt_notification::{Duration, Sound, Toast};
+use std::{process::exit, thread::sleep, time::Duration as StdDuration};
+
+use tauri_winrt_notification::{Duration, Sound, Toast, ToastNotification};
```
```suggestion
use tauri_winrt_notification::{Duration, Sound, Toast};
```

amrbashir committed:
Update examples/simple.rs

amrbashir commented:
Thank you for your contributions.
